lib/shoppingList.dart

```diff
@@ -0,0 +1,44 @@
+import 'package:flutter/material.dart';
+import 'products.dart';
+
+class MyShoppingList extends StatefulWidget {
+  List<TextEditingController> MyControllers = [];
+  MyShoppingList(List<TextEditingController> controllers)
+      : MyControllers = controllers;
+
+  @override
+  State createState() => _ShoppingListState(MyControllers);
+}
```

Comment:
Niepotrzebne przekazywanie
```suggestion
  @override
  State createState() => _ShoppingListState();
}
```